src/main/java/org/broadinstitute/hellbender/tools/walkers/contamination/GetPileupSummaries.java

```diff
@@ -0,0 +1,99 @@
+package org.broadinstitute.hellbender.tools.walkers.contamination;
+
+import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.vcf.VCFConstants;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.programgroups.VariantProgramGroup;
+import org.broadinstitute.hellbender.engine.FeatureContext;
+import org.broadinstitute.hellbender.engine.MultiVariantWalker;
+import org.broadinstitute.hellbender.engine.ReadsContext;
+import org.broadinstitute.hellbender.engine.ReferenceContext;
+import org.broadinstitute.hellbender.utils.GATKProtectedVariantContextUtils;
+import org.broadinstitute.hellbender.utils.pileup.ReadPileup;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Given a sites vcf of common SNPs and a bam file, get summary of alt and ref counts along with population allele frequency.
+ *
+ * The output of this tool is the input for {@link CalculateContamination}.  Use them as follows:
+ *
+ * java -jar gatk.jar GetPileupSummaries -V ExAC.vcf -V thousand_genomes.vcf -I input.bam -O pileups.table
+ * java -jar gatk.jar CalculateContamination -I pileups.table -O contamination.table
+ *
+ * Created by David Benjamin on 2/15/17.
+ */
+@CommandLineProgramProperties(
+        summary = "Calculate pileup statistics for inferring contamination",
+        oneLineSummary = "Calculate pileup statistics for inferring contamination",
+        programGroup = VariantProgramGroup.class)
+public class GetPileupSummaries extends MultiVariantWalker {
+
+    public final String MAX_SITE_AF_LONG_NAME = "maximumPopulationAlleleFrequency";
+    public final String MIN_SITE_AF_LONG_NAME = "minimumPopulationAlleleFrequency";
+    public final String MAX_SITE_AF_SHORT_NAME = "maxAF";
+    public final String MIN_SITE_AF_SHORT_NAME = "minAF";
+
+    private static final double DEFAULT_MIN_POPULATION_AF = 0.01;
+    private static final double DEFAULT_MAX_POPULATION_AF = 0.2;
```

Comment:
Back-of-the-envelope reasoning.  ExAC has about 6 million SNPs.  Most of them are very rare, and if you iterate over them the tool is too slow.  If you take only variants with allele fraction 0.01 and above, you will get about 1% of all ExAC variants, 6 x 10^4 sites.  Because most of these sites are closer to the low end of allele frequency (it's kind of a power law), let's wave our hands and say that these 6 x 10^4 sites all have allele frequency 0.05 = 5 x 10^-2.  The probability of being hom alt is the square of this.  Then we get about 6 x 25 x 10^4 x 10^-4 or about 150 hom vars in this range per exome.

Now if our estimate is based on the fraction of ref reads in these 150 sites, as long as the allele frequency is not too big the variance in the number of ref reads is roughly equal to the expectation (binomial variance is p(1-p)*n ~ pn if p is small).  Thus if our estimate of the contamination is (ref reads / total reads) and the depth is d and N = 150 is the number of hom alt sites, the variance is (contamination * Nd / (Nd)^2) = contamination/(Nd).  If the depth is 20 and the contamination is 10% for these 150 hom alt sites we get a standard deviation of sqrt(0.1 / (3000)) ~ 0.006, which is pretty good.  If we included rarer sites we would not get many more hom alts and the IO cost in run time would be huge.

As for the upper limit, it probably doesn't make that much of a difference.  It's probably superstitious, but I just like the idea that the estimate would still be good (i.e. only off by 20%) even if we naively assumed that all contamination was ref.  The non-superstitious defense of this is that common variants' allele frequency might vary a lot from population to population, throwing off the estimate, whereas rarer variants have less wiggle room.